src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -2853,6 +2853,11 @@ public void visitLambda(JCLambda tree) {
             int nextadrPrev = nextadr;
             ListBuffer<PendingExit> prevPending = pendingExits;
             try {
+                // JLS 16.1.10: No rule allows V to be definitely unassigned before a lambda
+                // body. This is by design: a variable that was definitely unassigned before the
+                // lambda body may end up being assigned to later on, so we cannot conclude that
+                // the variable will be unassigned when the body is executed.
+                uninits.excludeFrom(firstadr);
```

Comment:
I think this will exclude all tracked variables not only fields, shouldn't we exclude fields only?